core/src/main/scala/sttp/tapir/server/DecodeFailureContext.scala

```diff
@@ -2,4 +2,4 @@ package sttp.tapir.server
 
 import sttp.tapir.{DecodeResult, EndpointInput}
 
-case class DecodeFailureContext(input: EndpointInput[_], failure: DecodeResult.Failure)
+case class DecodeFailureContext(input: EndpointInput[_], failure: DecodeResult.Failure, rootInput: EndpointInput[_])
```

Comment:
isn't `rootInput` always `endpoint.input`? Maybe it would make sense to simply include the whole failing endpoint here. Then we could access the input, among others. What do you think?